sonar-db/src/main/resources/org/sonar/db/component/ComponentMapper.xml

```diff
@@ -250,6 +250,40 @@
     ORDER BY UPPER(p.name), p.name
   </select>
 
+  <select id="selectByQuery" resultType="Component">
+    select
+    <include refid="componentColumns"/>
+    <include refid="sqlSelectByQuery" />
+    ORDER BY LOWER(p.name), p.name
+  </select>
+
+  <select id="countByQuery" resultType="int">
+    select count(p.id)
+    <include refid="sqlSelectByQuery" />
+  </select>
+
+  <sql id="sqlSelectByQuery">
+    from projects p
+    <if test="nameOrKeyQuery!=null">
+      inner join resource_index ri on p.id = ri.resource_id
+    </if>
+    <where>
+      AND p.enabled=${_true}
```

Comment:
I could, I find easier to separate the different parts when looking at the xml file